### PR TITLE
Added Emulator CI and additional tests for Azure Cosmos SDK

### PR DESCRIPTION
**Pull request by kushagraThapar**
# Description:

* Current query (both sync and async) tests are running on single partition collections. 
* Duplicated the same query tests to run on multiple partition collection for sync and async APIs.
* Fixed setup and cleanup process of tests to make sure tests can run in parallel without causing resource conflicts.
* Fixed some of the skipped tests and enabled them
* Fixed Cosmos CI to run on Emulator by default
* Removed Cosmos Emulator windows CI - for python 3.11 - because of this issue - https://github.com/Azure/azure-sdk-for-python/issues/32409
* Fixed max parallel on cosmos live tests.
* Fixed database / container cleanups using pytest fixtures.

### CODE REVIEW

File: sdk/cosmos/cosmos-emulator-matrix.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "PythonVersion": 3.9,
+            "CoverageArg": "--disablecov",
+            "TestSamples": "false",
+            "ToxTestEnv": "'whl,sdist'",
+            "ACCOUNT_HOST": "https://localhost:8081/",
+            "ACCOUNT_KEY": "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
+            "Skip.Analyze": "true"
+        },
+        "Emulator Tests Python 3.10": {
+            "PythonVersion": 3.10,
```
Comment (scbedd) [inline sdk/cosmos/cosmos-emulator-matrix.json:31]:
```suggestion
            "PythonVersion": "3.10",
```

Somewhere along the line the `0` is getting stripped. We've only ever tested this against string values for python version, given that `3.10.0` (if you specified patch version) isn't a number that is parsable. A lot of effort to auto-cast when we could just surround in quotes 👍

**Comment by simorenoh** [comment]
Still going through the files but figured I'd leave two comments on things I saw.

Firstly, I see that the use of the `unittest` package has been re-added to the SDK tests in several places - we had recently moved away from using that package since it was supposed to have issues with running async tests and most/all other Python SDKs are no longer using it in favor of the default `pytest` package and the `test proxy` that Core offers (that we purposely don't use since we don't want to run tests against recordings.) Here was the PR: https://github.com/Azure/azure-sdk-for-python/pull/32076 - I believe to be aligned with everyone else we should probably not re-add this package unless we absolutely need it for some reason.

Secondly, more of a nitpick, but I see in several files there is not a blank line at the end of the files - which I believe is a formatting rule that we try to use in all our SDKs (I believe even Java does the same).

So far so good with the actual changes though, and it does seem like the pipelines are all passing 👍 thanks Kushagra!

**Comment by kushagraThapar** [comment]
@simorenoh - understood, however, `unittest` only had issues with running async tests because of python 3.7. Starting python 3.8 it can run both sync and async just fine. Currently I am using both `unittest` and `pytest`. 
`unittest` is great for standard testing practices as it provides boiler plate code for `setup`, `teardown`, `setupClass` and `tearDownClass` whereas `pytest` is used for fixtures which are used across the whole test setup. In my opinion, pytest is more advanced and might be an overkill in some of the basic test setup.

Regarding new line at the end, I will add one.

File: sdk/cosmos/azure-cosmos/test/test_session.py [hunk 2]
```diff
@@ -2,47 +2,53 @@
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/test/test_session.py:2]: marking this to add the MIT disclaimer as well

File: sdk/cosmos/azure-cosmos/test/test_session_token_unit.py [hunk 3]
```diff
@@ -1,15 +1,17 @@
 import unittest
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/test/test_session_token_unit.py:1]: also here

File: sdk/cosmos/azure-cosmos/test/test_streaming_failover.py [hunk 4]
```diff
@@ -1,26 +1,24 @@
 import unittest
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/test/test_streaming_failover.py:1]: also here

**Review by simorenoh — COMMENTED** [review]
ready to approve when we fix that one file with the imports splitting the MIT disclaimer, thanks so much Kushagra! 👍

**Review by simorenoh — APPROVED** [review]

**Pull request merged**